### ISSUE TITLE
Bad payload received by generic incoming webhook

### ISSUE THREAD

**Issue by GeorgeCimpoies** [waigel/node-ms-teams-webhook#10]
Hi, about a week ago I discovered that my previously working teams notifications are not being received anymore, and upon investigation I found that version 2.0.1 seems to be affected by the following error:

```
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'aib5.webhook.office.com',
      protocol: 'https:',
      [Symbol(kCapture)]: false,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(kEndCalled)]: true,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype] {
        accept: [ 'Accept', 'application/json, text/plain, */*' ],
        'content-type': [ 'Content-Type', 'application/x-www-form-urlencoded' ],
        'user-agent': [ 'User-Agent', 'axios/0.21.4' ],
        'content-length': [ 'Content-Length', 608 ],
        host: [ 'Host', 'aib5.webhook.office.com' ]
      },
      [Symbol(kUniqueHeaders)]: null
    },
    data: 'Bad payload received by generic incoming webhook.'
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
```

when using the simple example as below:

```
async function sendTeamsNotification(payload) {
    const url = process.env.MS_TEAMS_WEBHOOK_URL
    const webhook = new IncomingWebhook(url)

    await webhook.send(JSON.stringify(payload))
}
```

The payload is a simple MS Teams Card which was validated via https://messagecardplayground.azurewebsites.net/
The same request works via cURL:
`curl -H 'Content-Type: application/json' -d 'PAYLOAD' WEBHOOK`

From the error above, notice the following:
`'content-type': [ 'Content-Type', 'application/x-www-form-urlencoded' ],`
I was able to replicate the error via curl by changing the content-type from json to urlencoded.

Judging by this, we need to be able to specify content-type or to figure out why it is sent as urlencoded

**Comment by waigel** [comment]
Hi Matt,

you are right, if you stringify the JSON object you will get the wrong content type error. But adding a content header doesn't fix the problem. Is there any particular reason why you need to stringify the JSON object? In version 2.0.x, you can use the JSON object directly.

This example for me: 
```js
const { IncomingWebhook } = require("ms-teams-webhook");

async function sendTeamsNotification(payload) {
  const url = process.env.MS_TEAMS_WEBHOOK_URL;
  const webhook = new IncomingWebhook(url);
  await webhook.send(payload);
}

sendTeamsNotification({
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  summary: "Issue 176715375",
  themeColor: "0078D7",
  title: 'Issue opened: "Push notifications not working"',
  sections: [
    {
      activityTitle: "Miguel Garcie",
      activitySubtitle: "9/13/2016, 11:46am",
      activityImage:
        "https://connectorsdemo.azurewebsites.net/images/MSC12_Oscar_002.jpg",
      text: "There is a problem with Push notifications, they don't seem to be picked up by the connector.",
    },
  ],
});
```
In typescript we have also prohibited the string input. Does that help you?

**Comment by waigel** [comment]
I have released version `2.0.3` which validates the input and throws an error if it is a string.

**Comment by GeorgeCimpoies** [comment]
Wow, that was quick! Thank you @waigel! I remember I added the stringify as it was required in previous versions, but I hereby confirm that the payload can be used directly with version 2.0.1
